src/ol/View.js

```diff
@@ -365,6 +365,18 @@ class View extends BaseObject {
      */
     this.viewportSize_ = [100, 100];
 
+    /**
+     * @private
+     * @type {boolean} Whether the viewport has been set externally or if it is set to the default value.
+     */
+    this.viewportSizeSet_ = false;
+
+    /**
+     * @private
+     * @type {Array<() => void>} List of promise resolve functions waiting for the viewport to be set.
+     */
+    this.deferredFitPromiseResolves_ = [];
```

Comment:
A single promise should be enough here - the same unresolved promise can be returned for subsequent `fit()` calls.